README.md

```diff
@@ -1 +1,85 @@
-# TreasureData
\ No newline at end of file
+# TreasureData #
+
+This library allows your agent code to send data to the [Treasure Data platform](https://www.treasuredata.com) via the [Treasure Data REST API](https://support.treasuredata.com/hc/en-us/articles/360000675487-Postback-API).
+
+**To add this library to your project, add** `#require "TreasureData.agent.lib.nut:1.0.0"` **to the top of your agent code**
+
+## Library Usage ##
+
+### Prerequisites ###
+
+Before using the library you need to have:
+
+- A [Treasure Data API key](https://support.treasuredata.com/hc/en-us/articles/360000763288-Get-API-Keys). For data sending operations the Write-only key is enough.
+- Preconfigured [Treasure Data database and table](https://support.treasuredata.com/hc/en-us/articles/360001266348-Database-and-Table-Management).
+
+### Examples ###
+
+Working examples are provided in the [Examples](./Examples) directory.
+
+### Callbacks ###
```

Comment:
Heading here are different than our usual library readme's.

as 2nd level headers we usually have:
"Class Usage" that contains constructor and callback sections
"Class Methods" descriptions for each method

See here for a very similar library: https://github.com/electricimp/Losant

Reply:
I've never seen a fixed template for a readme. But I believe most of our libs have the following general structure in the readme:
- Library Usage: prerequisites, general explanation, design and usage details, howtos, examples, etc. - exact internals depend on a particular lib.
- "Library API Spec" (unfortunately, called differently in different readmes): a formal specification of the lib classes and methods.

Let's maybe elaborate more on a recommended (but not strictly mandated) template for a readme. Inviting Tony as well. If you OK, I will formulate my proposal by email later.

As for "Class Usage" and "Class Methods" - I personally do not see any benefits from these subsections.
In my opinion such a simple structure would be enough for the spec part:
"Library API Spec" 
  Class A
    Constructor A
    Method X
      Callback Y
    Method Z
  Class B
    Constructor B
    Method X
    Method Z
      Callback Y
...

Reply:
No problem - I'll leave this for Tony to sort out.